extensions-core/hdfs-storage/src/test/java/io/druid/indexing/common/tasklogs/HdfsTaskLogsTest.java

```diff
@@ -68,17 +68,47 @@ public void testOverwrite() throws Exception
 
     Files.write("blah", logFile, Charsets.UTF_8);
     taskLogs.pushTaskLog("foo", logFile);
-    Assert.assertEquals("blah", readLog(taskLogs, 0));
+    Assert.assertEquals("blah", readLog(taskLogs, "foo", 0));
 
     Files.write("blah blah", logFile, Charsets.UTF_8);
     taskLogs.pushTaskLog("foo", logFile);
-    Assert.assertEquals("blah blah", readLog(taskLogs, 0));
+    Assert.assertEquals("blah blah", readLog(taskLogs, "foo", 0));
   }
 
-  private String readLog(TaskLogs taskLogs, long offset) throws IOException
+  @Test
+  public void testKill() throws Exception
+  {
+    final File tmpDir = tempFolder.newFolder();
+    final File logDir = new File(tmpDir, "logs");
+    final File logFile = new File(tmpDir, "log");
+
+    final TaskLogs taskLogs = new HdfsTaskLogs(new HdfsTaskLogsConfig(logDir.toString()), new Configuration());
+
+    Files.write("log1content", logFile, Charsets.UTF_8);
+    taskLogs.pushTaskLog("log1", logFile);
+    Assert.assertEquals("log1content", readLog(taskLogs, "log1", 0));
+
+    //File modification timestamp is only maintained to seconds resolution, so artificial delay
+    //is necessary to separate 2 file creations by a timestamp that would result in only one
+    //of them getting deleted
+    Thread.sleep(1500);
+    long time = (System.currentTimeMillis()/1000)*1000;
+
+    Files.write("log2content", logFile, Charsets.UTF_8);
+    taskLogs.pushTaskLog("log2", logFile);
+    Assert.assertEquals("log2content", readLog(taskLogs, "log2", 0));
+
+    taskLogs.kill(time);
```

Comment:
Before here, can you check and make sure the modified time for the old file is before `time` and the new file is after `time`?